nixos/modules/tasks/auto-upgrade.nix

```diff
@@ -82,11 +98,23 @@ let cfg = config.system.autoUpgrade; in
 
   config = lib.mkIf cfg.enable {
 
-    system.autoUpgrade.flags =
-      [ "--no-build-output" ]
-      ++ (if cfg.channel == null
-          then [ "--upgrade" ]
-          else [ "-I" "nixpkgs=${cfg.channel}/nixexprs.tar.xz" ]);
+    assertions = [{
+      assertion = !((cfg.channel != null) && (cfg.flake != null));
+      message = ''
+        Updating nixos with the use of flakes cannot use channels.
```

Comment:
Probably better to be explicit, e.g. `The options 'system.autoUpgrade.channels' and 'system.autoUpgrade.flake' cannot both be set.`.

Reply:
used your example